src/lifecycle/props-init.js

```diff
@@ -96,13 +95,16 @@ function createNativePropertyDefinition(name, opts) {
     const attributeName = data(this, 'propertyLinks')[name];
     if (attributeName && !propData.settingAttribute) {
       const serializedValue = opts.serialize(newValue);
+      const currentAttrValue = this.getAttribute(attributeName);
+      const attributeChanged = !((empty(serializedValue) && empty(currentAttrValue)) ||
+                                (serializedValue === currentAttrValue));
       propData.syncingAttribute = true;
       if (shouldRemoveAttribute || empty(serializedValue)) {
```

Comment:
You can probably store the check to `empty(serializedValue)` since it's used more than once.